internal/controller/utils.go

```diff
@@ -45,6 +45,22 @@ func setDeploymentReplicas(deployment *appsv1.Deployment, replicas int32) bool {
 	return false
 }
 
+func setTolerations(deployment *appsv1.Deployment, tolerations []corev1.Toleration) bool {
+	if !apiequality.Semantic.DeepEqual(deployment.Spec.Template.Spec.Tolerations, tolerations) {
+		deployment.Spec.Template.Spec.Tolerations = tolerations
+		return true
+	}
+	return false
+}
+
+func setNodeSelector(deployment *appsv1.Deployment, nodeSelector map[string]string) bool {
+	if !apiequality.Semantic.DeepEqual(deployment.Spec.Template.Spec.NodeSelector, nodeSelector) {
```

Comment:
dtto

Reply:
Yes , the object is different than what is present , we update.